apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -177,7 +195,9 @@ defmodule GameBackend.CurseOfMirra.Quests do
         arena_match_results: arena_match_results,
         user: %User{daily_quests: daily_quests}
       }) do
-    Enum.reduce(daily_quests, [], fn daily_quest, acc ->
+    daily_quests
+    |> Enum.filter(fn daily_quest -> daily_quest.quest.type == "daily" end)
+    |> Enum.reduce([], fn daily_quest, acc ->
```

Comment:
Why not add this to the filter?

daily quest is "daily" and daily quest is completed

Reply:
It makes sense yeah! done